File: October/Week2/Day1/binarySearch.php
class Solution {

/**
 * @param Integer[] $nums
 * @param Integer $target
 * @return Integer
 */

function search($nums, $target) {
    $low = 0;
    $high = count($nums)-1;
    
    // return binarySearch($nums, $low, $high, $target);
    
    // echo $low, $high;
    while ($high >= $low) {
        $mid = ($low + $high)/2;
        
        if ($nums[$mid] == $target) {
            return floor($mid);
        }
        
        if ($nums[$mid] > $target) {
            $low = $mid+1;
        }
        
        if ($nums[$mid] < $target) {
            $high = $mid-1;
        }
    }
    
    return -1;
    // echo $low;
}
}

File: October/Week2/Day1/binarySearch.c


int binarySearch(int nums[], int l, int h, int k) {
    if (h >= l) {
        int mid = (l+h)/2;
        
        if (nums[mid] == k) {
            return mid;
        }
        
        if (nums[mid] > k) {
            return binarySearch(nums,l,mid-1,k);
        }
        
        if (nums[mid] < k) {
            return binarySearch(nums,mid+1,h,k);
        }
    }
    
    return -1;
}
int search(int* nums, int numsSize, int target){
    int low = 0;
    int high = numsSize;
    
    return binarySearch(nums, low, high-1, target);
}